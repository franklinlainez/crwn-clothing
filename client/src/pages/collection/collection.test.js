import { shallow } from "enzyme";
import CollectionPage from "./collection.component";
import CollectionItem from "../../components/collection-item/collection-item.component";

const mockCollectionItems = {
  title: "Hats",
  items: [
    {
      id: 1,
    },
    {
      id: 2,
    },
    {
      id: 3,
    },
  ],
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn(() => ({ collectionId: 69 })),
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(() => mockCollectionItems),
}));

describe("Collection component", () => {
  let wrapper = shallow(<CollectionPage />);

  it("render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("render the same number of collection as collection array", () => {
    expect(wrapper.find(CollectionItem).length).toBe(
      mockCollectionItems.items.length
    );
  });
});
