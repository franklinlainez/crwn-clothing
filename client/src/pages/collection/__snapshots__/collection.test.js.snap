// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection component render component 1`] = `ShallowWrapper {}`;
